artemis/reporting/modules/vcs/reporter.py

```diff
@@ -0,0 +1,239 @@
+        if task_result["headers"]["receiver"] != "vcs":
+            return []
+
+        if not task_result["status"] == "INTERESTING":
+            return []
+
+        assert not (
+            task_result["result"]["svn"] and task_result["result"]["git"]
+        ), "Found a suspicious case where both correct SVN and Git repositories are present."
+
+        if task_result["result"]["svn"]:
+            return VCSReporter._create_reports_svn(task_result, language)
+
+        if task_result["result"]["git"]:
+            return VCSReporter._create_reports_git(task_result, language)
+
+        return []
+
+    @staticmethod
+    def _create_reports_svn(task_result: Dict[str, Any], language: Language) -> List[Report]:
+        repo_url = urllib.parse.urljoin(get_target(task_result), ".svn/")
+
+        db_url = urllib.parse.urljoin(repo_url, "wc.db")
+
+        logger.info("Analysing SVN folder in %s", repo_url)
+        try:
+            with tempfile.NamedTemporaryFile() as f:
+                data = cached_get(db_url, max_size=config.Config.VCS_MAX_DB_SIZE_BYTES).content_bytes
+
+                f.write(data)
+                f.flush()
+                os.fsync(f.fileno())
+
+                con = sqlite3.connect(f.name)
+                cur = con.cursor()
+                cur.execute("select root from REPOSITORY")
+                (remote_url,) = cur.fetchone()
+        except Exception:
+            logger.exception("Unable to obtain repository remote url for %s", repo_url)
+            return []
+
+        if remote_url.startswith("http://") or remote_url.startswith("https://"):
+            parsed_remote_url = urllib.parse.urlparse(remote_url)
+            if parsed_remote_url.password:
+                return [
+                    Report(
+                        top_level_target=get_top_level_target(task_result),
+                        target=repo_url,
+                        report_type=VCSReporter.EXPOSED_VERSION_CONTROL_FOLDER_WITH_CREDENTIALS,
+                        report_data={
+                            "config_url": db_url,
+                            "username": parsed_remote_url.username,
+                            "password_prefix": parsed_remote_url.password[:3],
+                        },
+                        timestamp=task_result["created_at"],
+                    )
+                ]
+
+        logger.info("Successfully obtained remote url: %s", remote_url)
+
+        try:
+            with tempfile.TemporaryDirectory() as output_dir_name:
+                subprocess.check_call(
+                    ["svn", "co", remote_url, "--depth", "empty", output_dir_name],
+                    stdin=subprocess.DEVNULL,
+                    stdout=subprocess.DEVNULL,
+                    stderr=subprocess.DEVNULL,
+                    timeout=30,
+                )
+            logger.info("Successfully cloned %s, that means it's a public repo - not reporting", remote_url)
```

Comment:
Well, that's just naive.

Reply:
yup, that's a heuristic to distinguish things worth reporting from not worth reporting. That works for us. I you have a better approximation, let's discuss it - do you have?